plugins/modules/mso_schema_site_contract_service_graph_listener.py

```diff
@@ -0,0 +1,744 @@
+                elif (listener_protocol == "http" and health_check.get("protocol") == "https") or (
+                    listener_protocol == "https" and health_check.get("protocol") == "http"
+                ):
+                    mso.input_validation(
+                        "health_check - 'protocol'", "http/https", ["port", "path", "unhealthy_threshold", "timeout", "interval"], health_check
+                    )
+                else:
+                    mso.fail_json(
+                        msg=(
+                            "The 'listener_protocol': {0} and the health_check protocol: {1} "
+                            + "is not a valid configuration at the object position: {2} and the object name: {3}"
+                        ).format(listener_protocol, health_check.get("protocol"), position, rule.get("name"))
+                    )
+
+                health_check_data["port"] = health_check.get("port")
+                health_check_data["protocol"] = health_check.get("protocol")
+                health_check_data["path"] = health_check.get("path")
+                health_check_data["interval"] = health_check.get("interval")
+                health_check_data["timeout"] = health_check.get("timeout")
+                health_check_data["unhealthyThreshold"] = health_check.get("unhealthy_threshold")
+                health_check_data["successCode"] = health_check.get("success_code")
+                health_check_data["useHostFromRule"] = YES_OR_NO_TO_BOOL_STRING_MAP.get(health_check.get("use_host_from_rule"))
+                health_check_data["host"] = health_check.get("host")
+
+                rule_data["healthCheck"] = health_check_data
+
+            rules_data.append(rule_data)
+
+        listener_object["rules"] = rules_data
+
+        if parent_present:
+            # Update an existing listener
+            if mso.existing:
+                listener_path = (
+                    "/sites/{0}-{1}/contracts/{2}/serviceGraphRelationship/serviceNodesRelationship/{3}"
+                    + "/deviceConfiguration/cloudLoadBalancer/listeners/{4}"
+                ).format(site_id, template, contract, service_node_index, listener)
+                op = "replace"
+            else:
+                # Create a new listener
+                listener_path = (
+                    "/sites/{0}-{1}/contracts/{2}/serviceGraphRelationship/serviceNodesRelationship/{3}" + "/deviceConfiguration/cloudLoadBalancer/listeners/-"
+                ).format(site_id, template, contract, service_node_index)
+                op = "add"
+            parent_object = listener_object
+        else:
+            # Create a new listener with parent object
+            listener_path = "/sites/{0}-{1}/contracts/{2}/serviceGraphRelationship/serviceNodesRelationship/{3}".format(
+                site_id, template, contract, service_node_index
+            )
+            op = "replace"
```

Comment:
why is this a replace when something does not exist yet?

Reply:
Initially, the site contract contains only `serviceNodesRelationship` with a list of service node references.

To create a listener, we need to attach the `deviceConfiguration.cloudLoadBalancer.listeners` with the current parent object. So we need to update the existing object with the PATCH-replace operation. Also it depends on the `parent_present` flag value.